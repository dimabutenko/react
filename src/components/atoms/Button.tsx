import React, { type ReactNode, type MouseEventHandler } from 'react'

interface ButtonProps {
  class?: string | undefined
  onClick?: MouseEventHandler
  children: ReactNode
}

const Button: React.FC<ButtonProps> = (props: ButtonProps) => (
  <button className={props.class} onClick={props.onClick}>
    {props.children}
  </button>
)

export default Button
