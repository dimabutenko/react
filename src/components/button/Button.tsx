import React, { type ReactNode, type MouseEventHandler } from 'react'

interface ButtonProps {
  class?: string | undefined
  onClick?: MouseEventHandler
  children: ReactNode
  attributes?: any | undefined
}

const Button: React.FC<ButtonProps> = (props: ButtonProps) => (
  <button className={props.class} onClick={props.onClick} {...props.attributes}>
    {props.children}
  </button>
)

export default Button
