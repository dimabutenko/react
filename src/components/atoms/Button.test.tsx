import React from 'react'
import { render, screen } from '@testing-library/react'
import Button from './Button'
import * as userEvent from '@testing-library/user-event'

describe('Button', () => {
  test('Should render correctly', async () => {
    const { container } = render(<Button>Hello world</Button>)
    expect(container).toMatchSnapshot()
  })

  test('calls onDeleteClick when clicked', async () => {
    const user = userEvent.default.setup()
    const callback = jest.fn()
    render(<Button onClick={callback}>Hello world</Button>)
    await user.click(screen.getByText('Hello world'))
    expect(callback).toBeCalled()
  })
})
