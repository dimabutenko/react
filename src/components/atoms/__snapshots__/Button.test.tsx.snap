// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should render correctly 1`] = `
<div>
  <button>
    Hello world
  </button>
</div>
`;
